piper/buttondialog.py

```diff
@@ -0,0 +1,267 @@
+# Copyright (C) 2017 Jente Hidskes <hjdskes@gmail.com>
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+import sys
+
+from gettext import gettext as _
+
+from .gi_composites import GtkTemplate
+from .keystroke import KeyStroke
+
+import gi
+gi.require_version("Gtk", "3.0")
+from gi.repository import Gdk, GObject, Gtk
+
+
+@GtkTemplate(ui="/org/freedesktop/Piper/ui/ButtonDialog.ui")
+class ButtonDialog(Gtk.Dialog):
+    """A Gtk.Dialog subclass to implement the dialog that shows the
+    configuration options for button mappings."""
+
+    __gtype_name__ = "ButtonDialog"
+
+    _BUTTON_TYPE_TO_DESC = {
```

Comment:
I'm thinking of moving this to ratbagd.py; I'm going to have to add another dictionary like this for the special actions. Ideally, both of these will be in the bindings so they can more easily be kept in sync with libratbag.